src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.test.ts

```diff
@@ -0,0 +1,331 @@
+import { fixedPointAdjustment, toBNWei } from "../utils";
+import { MockUBAConfig } from "../clients/mocks";
+import { computePiecewiseLinearFunction } from "./UBAFeeUtility";
+
+describe("UBAFeeSpokeCalculatorAnalog", () => {
+  describe("getEventFee", () => {
+    // Create a chain Id that we can reference anywhere
+    // without any magic numbers.
+    const chainId = "1";
+
+    for (const decimalCount of [6, 12, 18]) {
+      describe(`Testing Decimals: ${decimalCount}`, () => {
+        // This is the base case for this function. We want to ensure that if we have a
+        // positive balancing fee result, that the result is equal to the integration of
+        // the balancing fee curve without any additional modifiers.
+        it("if balancing fee is positive, then the result is a single integration", () => {
+          // We should iterate for 1000 iterations to ensure that we have a good sample size
+          // for our fuzz testing.
+          for (let iteration = 1; iteration <= 1000; iteration++) {
+            // We set the amount, lastRunningBalance, multiplier, and lastIncentiveBalance for our test.
+            // Note: We can do this by generating random numbers
+            // Note: we need to ensure that this matches the fixed decimals of the balancing fee curve.
+            // Note: we need to ensure that the value is never 0 for these tests.
+            const amount = toBNWei(Math.floor(100 * Math.random()) + 1, decimalCount);
+            const lastRunningBalance = toBNWei(Math.floor(10_000 * Math.random() + 1), decimalCount);
+            const lastIncentiveBalance = toBNWei(Math.floor(10_000 * Math.random()) + 1, decimalCount);
+            const rewardMultiplier = utils.parseEther(Math.random().toFixed(18));
+
+            // We can establish a positive slope with a randomized curve.
+            const positiveSlope = toBNWei(Math.random().toFixed(18), 18);
+
+            // We start with a basic config that's been instantiated.
+            const config = new MockUBAConfig();
+            // We set the balancing fee curve to be a basic curve that has a positive slope.
+            // We can work in terms of any decimal as long as we're consistent & the second value
+            // in the tuple is in terms of wei or any other 18 decimal token.
+            config.setBalancingFeeCurve(chainId, [
+              [toBNWei(0, decimalCount), toBNWei(0, 0)],
+              [toBNWei(1, decimalCount), positiveSlope],
+            ]);
+            // We set the reward multiplier to be 1. This is to ensure that we don't have any
+            // additional reward multiplier that would affect our calculations.
+            config.setRewardMultiplier(chainId, rewardMultiplier);
+            // We call the getEventFee function with the parameters we've set above.
+            const fee = getEventFee(amount, "inflow", lastRunningBalance, lastIncentiveBalance, 1, config).balancingFee;
```

Comment:
is it worth testing outflow too or does that not increase test coverage?

Reply:
The difference between `inflow` and `outflow` is designed to be the direction of the integral. It shouldn't affect code coverage as it only appears in `getEventFee` as 

```
  let balancingFee = computePiecewiseLinearFunction(
    flowCurve,
    lastRunningBalance,
    lastRunningBalance.add(amount.mul(flowType === "inflow" ? 1 : -1))
  );

```